Libplanet.Net/Transports/NetMQTransport.cs

```diff
@@ -893,6 +890,15 @@ private async Task ProcessRequest(MessageRequest req, CancellationToken cancella
             {
                 tcs.TrySetException(e);
             }
+            catch (Exception ae)
+            {
+                var se = new SendMessageFailException(
+                    $"Unexpected exception occurred during {nameof(ProcessRequest)}().",
+                    req.Peer,
+                    ae
+                );
+                tcs.TrySetException(se);
+            }
```

Comment:
Now that we know more about `GetRequestDealerSocket()`, I think it is no longer an "unexpected exception". Perhaps wrapping `NetMQException` inside `SendMessageFailException` should also be done inside `GetRequestDealerSocket()` (instead of throwing `MetMQException` directly) and be caught by `catch when` statement above.

Reply:
Well... I felt that we might have to wrap all exceptions as `SendMessageFailExceptionn` (at least on the current structure) since it will hang again if there is any unexpected exception. (actually, we've faced `System.Net.Internals.SocketExceptionFactory+ExtendedSocketException ` even on #2424).

Of course, this may be also a signal that refactorings are needed. but considering that Libplanet is a library that is already used in production, it is also necessary to take a conservative approach to errors of known patterns (i.e., hangs when unexpected errors occur).

Reply:
Very well then, we can postpone it I suppose.